src/core/Tags.java

```diff
@@ -637,4 +637,19 @@ public static boolean looksLikeInteger(final String value) {
     return true;
   }
 
+  /**
+   * Returns true if the given string can fit into a float.
+   * @param value The String holding the float value.
+   * @return true if the value can fit into a float, false otherwise.
+   * @throws NumberFormatException if the value is not numeric.
+   * @since 2.0.2
+   */
+  public static boolean fitsInFloat(final String value) {
+    final float f = Float.parseFloat(value);
+    final String converted = Float.toString(f);
+
+    // this will be false if there was a loss of precision.
+    return value.equals(converted);
```

Comment:
How about reading straight to double, then taking the absolute value of this number and check it against `Float.MAX_VALUE` ? If it's greater then we keep it as double, otherwise we cast it down to float.